keanu-project/src/main/java/io/improbable/keanu/vertices/Vertex.java

```diff
@@ -1,21 +1,28 @@
 package io.improbable.keanu.vertices;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+import com.google.common.collect.ImmutableList;
+
 import io.improbable.keanu.algorithms.graphtraversal.DiscoverGraph;
 import io.improbable.keanu.algorithms.graphtraversal.VertexValuePropagation;
 import io.improbable.keanu.tensor.Tensor;
 import io.improbable.keanu.tensor.dbl.DoubleTensor;
 import io.improbable.keanu.vertices.dbl.KeanuRandom;
 
-import java.util.*;
-import java.util.concurrent.atomic.AtomicLong;
-
 public abstract class Vertex<T> {
 
     public static final AtomicLong ID_GENERATOR = new AtomicLong(0L);
 
     private long uuid = ID_GENERATOR.getAndIncrement();
-    private Set<Vertex> children = new HashSet<>();
-    private Set<Vertex> parents = new HashSet<>();
+    private List<Vertex> children = new ArrayList<>();
```

Comment:
Why are these `List`s rather than `Set`s? I can't see anything that specifies ordering in here, so a `set` seems more descriptive.

Reply:
This is pre-empting a change I've made on another feature branch that requires it (a Vertex backed by a Distribution needs to know how its parent vertices correspond to its parameters like mu and sigma)

Reply:
isn't that why we have `getMu()` and `getSigma()`?

I really wouldn't rely on ordering in a list as it'll be non-obvious that the order matters when it's constructed.

Reply:
I'm going to revert this for now, since it's not actually relevant to the Differentiable interface. I will revisit later since I'm going to change how we create the Distribution-backed Vertices anyway.